### ISSUE TITLE
`unused_setter_value` incorrectly triggers for overridden properties.

### ISSUE THREAD

**Issue by Coeur** [realm/SwiftLint#2585]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`unused_setter_value` incorrectly triggers for overridden properties.

I use the following code that I gave as [a Stack Overflow answer](https://stackoverflow.com/questions/27988279/uitextview/49428307#49428307):
```swift
/// sub-class to disallow selection.
class UnselectableTappableTextView: UITextView {
    // required to prevent blue background selection from any situation
    override var selectedTextRange: UITextRange? {
        get { return nil }
        // This triggers a SwiftLint violation:
        set {}
    }
}
```

Having an empty setter is the desired behavior when overriding.

### Environment

* SwiftLint version 0.30.0
* Xcode 10.1
* Swift 4.2

**Comment by jpsim** [comment]
Yeah I agree we should allow no-op `set`s for overridden variables.

**Comment by jgavris** [comment]
+1, Similarly for `@available(*, unavailable)` properties.

<img width="952" alt="screen shot 2019-01-25 at 2 23 31 pm" src="https://user-images.githubusercontent.com/178749/51768349-0f069a00-20ae-11e9-973d-00a5b0e03589.png">

**Comment by marcelofabri** [comment]
Note: you can silence the warning:

```swift
override var selectedTextRange: UITextRange? {
        get { return nil }
        set { _ = newValue }
    }
```

**Comment by jpsim** [comment]
That's a good workaround for now @marcelofabri although I think we should also implement this change because users shouldn't adapt their code just to satisfy SwiftLint unless it's also better for other reasons. I think in this case, an empty `set` sufficiently conveys the author's intention to not do anything with `newValue`.

**Comment by marcelofabri** [comment]
yeah, but this will be a warning on Swift too soon: https://github.com/apple/swift/pull/21983 and https://bugs.swift.org/browse/SR-964

I don't mind changing the behavior though, so PRs are welcome!

**Comment by username0x0a** [comment]
This change appears nice, but it would be good to have an option (attribute?) to allow enforcing this rules on `override`s as well as that might be the actual rule decision our team took for example, we're used to use the `_ = newValue` workaround when it's really desired, yet it can be still useful on overrides wrapping something else on inherited instances.

**Comment by Coeur** [comment]
For readers, to clarify, even though it's labeled as "_wontfix_", it got fixed on  21 Jan 2022 with #3653, three years after the issue was opened. 🎉 
It got released in SwiftLint [0.46.1](https://github.com/realm/SwiftLint/releases/tag/0.46.1) the next day.
